app/code/Magento/Quote/Model/QuoteManagement.php

```diff
@@ -354,6 +354,12 @@ public function placeOrder($cartId, PaymentInterface $paymentMethod = null)
         if ($quote->getCheckoutMethod() === self::METHOD_GUEST) {
             $quote->setCustomerId(null);
             $quote->setCustomerEmail($quote->getBillingAddress()->getEmail());
+            $quote->setCustomerFirstname($quote->getBillingAddress()->getFirstname());
+            $quote->setCustomerLastname($quote->getBillingAddress()->getLastname());
+
+            if ($quote->getBillingAddress()->getMiddlename()) {
+                $quote->setCustomerMiddlename($quote->getBillingAddress()->getMiddlename());
```

Comment:
Maybe set them just when they are equal to `null`?

Reply:
hmm, yes it would be better. ok

Reply:
All done. also fixed the failed code style test